Client/Frontend/Browser/LongPressGestureRecognizer.swift

```diff
@@ -43,64 +42,41 @@ class LongPressGestureRecognizer: UILongPressGestureRecognizer, UIGestureRecogni
         return true
     }
 
+    func gestureRecognizer(gestureRecognizer: UIGestureRecognizer, shouldBeRequiredToFailByGestureRecognizer otherGestureRecognizer: UIGestureRecognizer) -> Bool {
+        // Hack to detect the built-in context menu gesture recognizer.
+        return otherGestureRecognizer is UILongPressGestureRecognizer && otherGestureRecognizer.delegate?.description.rangeOfString("WKContentView") != nil
+    }
+
     // MARK: - Long Press Gesture Handling
     func SELdidLongPress(gestureRecognizer: UILongPressGestureRecognizer) {
         if gestureRecognizer.state == UIGestureRecognizerState.Began {
             //Finding actual touch location in webView
-            var touchLocation = gestureRecognizer.locationInView(self.browser.webView)
-            touchLocation.x -= self.browser.webView.scrollView.contentInset.left
-            touchLocation.y -= self.browser.webView.scrollView.contentInset.top
-            touchLocation.x /= self.browser.webView.scrollView.zoomScale
-            touchLocation.y /= self.browser.webView.scrollView.zoomScale
-
-            self.browser.webView.evaluateJavaScript("findElementsAtPoint(\(touchLocation.x),\(touchLocation.y))", completionHandler:nil)
+            var touchLocation = gestureRecognizer.locationInView(self.webView)
+            touchLocation.x -= self.webView.scrollView.contentInset.left
+            touchLocation.y -= self.webView.scrollView.contentInset.top
+            touchLocation.x /= self.webView.scrollView.zoomScale
+            touchLocation.y /= self.webView.scrollView.zoomScale
+
+            self.webView.evaluateJavaScript("findElementsAtPoint(\(touchLocation.x),\(touchLocation.y))") { (response: AnyObject!, error: NSError!) in
```

Comment:
I understand. Some time in the future we should experiment with doing the long tap detection in JS I think.